lib/src/platform/src/realtime/realtime.dart

```diff
@@ -37,6 +36,16 @@ class Realtime extends PlatformObject {
       AblyMessage(options),
     );
     _realtimeInstances[handle] = this;
+
+    if (io.Platform.isAndroid && options.autoConnect != false) {
```

Comment:
I don't like `options.autoConnect != false)`, but didn't want to add more complexity to this PR. This is done because `options.autoConnect` is nullable, it shouldn't be. Then it will be non null, we can just do `if (io.Platform.isAndroid && options.autoConnect) {`. See https://github.com/ably/ably-flutter/issues/230

Gonna leave this conversation here indefinitely without resolving it so reviewers understand it is temporary.